frappe/database/query.py

```diff
@@ -173,6 +173,9 @@ def _apply_filter(
 		if isinstance(_value, str) and has_function(_value):
 			_value = self.get_function_object(_value)
 
+		if isinstance(_value, (list, tuple)) and not _value:
+			_value = ("",)
```

Comment:
We are assuming that the user meant this. Also postgres won't like type-mismatches like these. 

In most cases user wont knowingly pass empty list, the list itself is result of some query or operation that yielded nothing. 

Also no test?

Reply:
If you pass an empty list, it generates an invalid query. There are legit places in code that depend on this behavior, I don't like it either.